packages/blockly-field-slider/src/field_slider.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Number slider input field.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+import * as Blockly from 'blockly/core';
+
+export class FieldSlider extends Blockly.FieldNumber {
+  /**
+   * Class for an number slider field.
+   * @param {string|number=} opt_value The initial value of the field. Should
+   *    cast to a number. Defaults to 0.
+   * @param {?(string|number)=} opt_min Minimum value.
+   * @param {?(string|number)=} opt_max Maximum value.
+   * @param {?(string|number)=} opt_precision Precision for value.
+   * @param {?Function=} opt_validator A function that is called to validate
+   *    changes to the field's value. Takes in a number & returns a validated
+   *    number, or null to abort the change.
+   * @param {Object=} opt_config A map of options used to configure the field.
+   *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/number#creation}
+   *    for a list of properties this parameter supports.
+   * @extends {Blockly.FieldNumber}
+   * @constructor
+   */
+  constructor(opt_value, opt_min, opt_max, opt_precision,
+      opt_validator, opt_config) {
+    super(opt_value, opt_min, opt_max, opt_precision, opt_validator, opt_config);
+
+    /**
+     * Array holding info needed to unbind events.
+     * Used for disposing.
+     * Ex: [[node, name, func], [node, name, func]].
+     * @type {!Array.<Array<?>>}
+     * @private
+     */
+    this.boundEvents_ = [];
+
+    /**
+     * The HTML range input element.
+     * @type {?HTMLInputElement}
+     * @private
+     */
+    this.sliderInput_ = null;
+  }
+
+  /**
+   * Creates and shows the slider field's editor.
+   * @protected
+   * @override
+   */
+  showEditor_() {
+    super.showEditor_();
```

Comment:
FieldTextInput which is the closest parent implementing showEditor_ expects and optional mouse event and whether or not the input is quite as optional parameters. I think we should pass these up: 
```
 * @param {Event=} _opt_e Optional mouse event that triggered the field to open,
 *     or undefined if triggered programmatically.
 * @param {boolean=} opt_quietInput True if editor should be created without
 *     focus.  Defaults to false.
```

Reply:
On IOS this shows both an alert and a slider, perhaps we should be overriding ``showInlineEditor_`` instead (which is currently set as private in field_textinput..

Reply:
https://github.com/google/blockly/blob/master/core/field_textinput.js#L287

Reply:
I've added the missing parameters to the supercall.

Reply:
Also, added handling based on FieldAngle:
https://github.com/google/blockly/blob/master/core/field_angle.js#L265